src/Channels/Networking/TransportProvider.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Globalization;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Channels.Networking
+{
+    /// <summary>
+    /// Represents a transport that is able to establish connections from client to server,
+    /// or to start a listener to operate as a server
+    /// </summary>
+    public abstract class TransportProvider : IDisposable
+    {
+        private ChannelFactory channelFactory;
+
+        /// <summary>
+        /// Provides access to the channel factory for this provider
+        /// </summary>
+        /// <remarks>This factory is lazily instantiated</remarks>
+        protected ChannelFactory ChannelFactory => channelFactory ?? (channelFactory = new ChannelFactory());
```

Comment:
Not sure about the lazy creation. This probably shouldn't be exposed at all really. The `TransportProvider` has to decide how many `ChannelFactories` (like one per thread) it was to allocate.